10.0 shopinvader promotion rule

lmignon opened the pull request:
requires:
- [x] https://github.com/akretion/payment-gateway/pull/36
- [x] https://github.com/akretion/ak-odoo-incubator/pull/125

BE CAREFUL: The last commit must be removed once dependencies are merged

sebastienbeau committed:
[REF] refactor module due to change done in the based module

sebastienbeau committed:
[REF] refactor test and reuse test on base module

lmignon committed:
[FIX] shopinvader_image, shopinvader_payment: remove 'from openerp'

lmignon committed:
[MIG] shopinvader_promotion_rule to 10.0

lmignon committed:
[IMP] shopinvader: Avoid multiple search on SO.
Improve perfs by taking avdantage of the cache

lmignon committed:
[ADD] setup.py

lmignon committed:
[FIX] Avoid to nest view into view.
Indeed when data are loaded in the installation process the onchange_spec (list of field on which to play onchange) is computed from the default view. This computation try to follow the path for relation fields. That works fine if we only have 1 level under the field (tree). Nevertheless, if you also define a form  with relation field definig their own nested form, the computed path exceed the 2 level supported by onchange (my_relation_field.my_subrelation_field.field_name)
It's a bug into odoo but. Nevertheless in all cases we must avoid this kind of form definition since these are painfull to extends

lmignon committed:
[IMP] shopinvader: mute_logger into tests for errors

pedrobaeza commented:
@lmignon I have been analyzing the problem here as you told me, and indeed seems to not be the same problem as we have on the rest of OCA with reports, but I have no clue how to solve it. Sorry for not being more helpful.

lmignon commented:
Thank you @pedrobaeza for sure we'll find why but when? ;-)

lmignon committed:
[FIX] tests payment use metaclass

lmignon committed:
[FIX] refactor test to avoid mismatch between db cursor
Whith multiple inheritance we must be sure that setup are executed into
a predicatable way. Moreover we must avoid to mix into a test class
parent using SavepointTransactionCase and others using a test registry

lmignon committed:
[FIX] shopinvader: Use shopinvader group in place of connector group

lmignon committed:
[TO REMOVE] Use dev externals fixing bugs....

lmignon committed:
try to have ak-odoo-incubator in first place into the addon_path in place of the one from storage ...

sebastienbeau commented:
Thanks Laurent for all of your work I have merged your PR without the two last commit here : https://github.com/akretion/odoo-shopinvader/commit/733313db958286c7b6a982b142b02f410344a965
